src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/reference/Reference.vue

```diff
@@ -91,8 +93,27 @@ export default defineComponent({
       }
     },
     handleSubmit() {
-      this.alertStore.setWarningAlert("Submit does not work yet");
+
+      this.alertStore.setWarningAlert("User Accepted");
     },
+    async handleDecline() {
+
+      const currentStepFormId = this.wizardStore.currentStep.form.id;
+      const formRef = (this.$refs.wizard as typeof Wizard).$refs[currentStepFormId][0].$refs[currentStepFormId];
+      const { valid } = await formRef.validate();
+
+      if (!valid) {
+        this.alertStore.setFailureAlert("Please fill out all required fields");
+        return;
+      }
+
+      const reason = this.wizardStore.wizardData[this.wizardStore.wizardConfig.steps.decline.form.inputs.referenceDecline.id];
+      const result = await optOutReference(this.$route.params.token as string, reason as Components.Schemas.UnabletoProvideReferenceReasons);
+      console.log(result);
+      if (result.data) {
+          this.$router.push({ path: "/reference-submitted" });
+        }
+    }
```

Comment:
We can add a loader icon for when a user has hit the declined button? 

This was done for the submit in Applications.vue
https://github.com/bcgov/ECC-ECER/blob/cf0c28a0d5d09df62514e181bf81a654c62cb7ad/src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Application.vue#L71

Reply:
Should prevent double clicks for users trying to opt-out.